recipes/default.rb

```diff
@@ -37,13 +36,25 @@ class ::Chef::Recipe
 Chef::Log.debug("Squid listen_interface: #{listen_interface}")
 Chef::Log.debug("Squid ipaddress: #{ipaddress}")
 Chef::Log.debug("Squid netmask: #{netmask}")
-Chef::Log.debug("Squid version: #{squid_version}")
 Chef::Log.debug("Squid host_acls: #{host_acl}")
 Chef::Log.debug("Squid url_acls: #{url_acl}")
 Chef::Log.debug("Squid acls: #{acls}")
 
+ruby_block 'Detect squid version' do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    command = %(#{node['squid']['package']} -v | grep Version | sed 's/.*Version \\\(.\\..\\\).*/\\1/g' | tr -d '\n')
+    command_out = shell_out(command)
+    node.set['squid']['squid_version_detected'] = command_out.stdout.to_f
```

Comment:
node.set has been deprecated. If you want to persist data on the node use node.normal